frontend/src/features/leases/detail/LeasePages/deposits/DepositsContainer.tsx

```diff
@@ -0,0 +1,257 @@
+   */
+  const onSaveDeposit = async (depositForm: FormLeaseDeposit) => {
+    console.log('saving deposit!', depositForm);
+    if (lease && lease.id && lease.rowVersion) {
+      let request: IParentConcurrencyGuard<ILeaseSecurityDeposit> = {
+        parentId: lease.id,
+        parentRowVersion: lease.rowVersion,
+        payload: FormLeaseDeposit.toInterfaceModel(depositForm),
+      };
+      const updatedLease = await updateLeaseDeposit(request);
+      if (!!updatedLease?.id) {
+        setLease(updatedLease);
+        setEditDepositValue(FormLeaseDeposit.createEmpty());
+        setShowEditModal(false);
+      }
+    } else {
+      console.log('Lease information incomplete');
+    }
+  };
+
+  const onEditReturnDeposit = (id: number) => {
+    var deposit = depositReturns.find(x => x.id === id);
+    if (deposit) {
+      var parentDeposit = securityDeposits.find(x => x.id === deposit?.parentDepositId);
+      if (parentDeposit) {
+        setEditReturnValue(FormLeaseDepositReturn.createFromModel(deposit, parentDeposit));
+        setShowReturnEditModal(true);
+      } else {
+        console.log('Parent deposit incomplete');
+      }
+    }
+  };
+
+  const onDeleteDepositReturn = (id: number) => {
+    var deposit = depositReturns.find(x => x.id === id);
+    if (deposit) {
+      var parentDeposit = securityDeposits.find(x => x.id === deposit?.parentDepositId);
+      if (parentDeposit) {
+        setDepositReturnToDelete(FormLeaseDepositReturn.createFromModel(deposit, parentDeposit));
+        setDeleteReturnModalWarning(true);
+      } else {
+        console.log('Parent deposit incomplete');
+      }
+    }
+  };
+
+  /**
+   * Send the save request (either an update or an add). Use the response to update the parent lease.
+   * @param returnDepositForm
+   */
+  const onSaveReturnDeposit = async (returnDepositForm: FormLeaseDepositReturn) => {
+    console.log('saving return deposit!', returnDepositForm);
+    if (lease && lease.id && lease.rowVersion) {
+      let request: IParentConcurrencyGuard<ILeaseSecurityDepositReturn> = {
+        parentId: lease.id,
+        parentRowVersion: lease.rowVersion,
+        payload: FormLeaseDepositReturn.toInterfaceModel(returnDepositForm),
+      };
+      const updatedLease = await updateLeaseDepositReturn(request);
+      if (!!updatedLease?.id) {
+        setLease(updatedLease);
+        setDepositReturnToDelete(undefined);
+        setShowReturnEditModal(false);
+      }
+    } else {
+      console.log('Lease information incomplete');
+    }
+  };
+
+  return (
+    <Styled.DepositsContainer>
+      <DepositsReceivedContainer
+        securityDeposits={securityDeposits}
+        depositReturns={depositReturns}
+        onAdd={() => {
+          onAddDeposit();
+        }}
+        onEdit={(id: number) => {
+          onEditDeposit(id);
+        }}
+        onDelete={(id: number) => {
+          onDeleteDeposit(id);
+        }}
+        onReturn={(id: number) => {
```

Comment:
can you just do onReturn={onReturnDeposit}
The additional wrapper function seems unecessary.

Reply:
true, will make it more straight forward